sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -1365,3 +1366,38 @@ def estimate_size(self, value, nested=False):
     # type: (Any, bool) -> int
     value_size = self._value_coder.estimate_size(value)
     return get_varint_size(value_size) + value_size
+
+
+class ShardedKeyCoderImpl(StreamCoderImpl):
+  """For internal use only; no backwards-compatibility guarantees.
+
+  A coder for sharded user keys.
+
+  The encoding and decoding should follow the order:
+      length of shard id byte string
+      shard id byte string
+      encoded user key
+  """
+  def __init__(self, key_coder_impl):
+    self._shard_id_coder_impl = LengthPrefixCoderImpl(BytesCoderImpl())
+    self._key_coder_impl = key_coder_impl
+
+  def encode_to_stream(self, value, out, nested):
+    # type: (ShardedKey, create_OutputStream, bool) -> None
+    self._shard_id_coder_impl.encode_to_stream(value.shard_id, out, True)
+    self._key_coder_impl.encode_to_stream(value.key, out, True)
```

Comment:
```suggestion
    self._key_coder_impl.encode_to_stream(value.key, out, nested)
```

Reply:
Didn't change this in the latest commit since it would by default apply an unnested context which doesn't seem to be consistent with the Java encoding. Please see my other comment above.